src/components/ActivityCollector/attachClickActivityCollector.js

```diff
@@ -10,18 +10,25 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+import { stackError } from "../../utils";
+
 const createClickHandler = (eventManager, lifecycle) => {
   return clickEvent => {
     // TODO: Consider safeguarding from the same object being clicked multiple times in rapid succession?
     const clickedElement = clickEvent.target;
     const event = eventManager.createEvent();
-    lifecycle.onClick({ event, clickedElement }).then(() => {
-      if (event.isEmpty()) {
-        return;
-      }
+    return lifecycle
+      .onClick({ event, clickedElement })
+      .then(() => {
+        if (!event.isEmpty()) {
+          return Promise.resolve();
+        }
 
-      eventManager.sendEvent(event);
-    });
+        return eventManager.sendEvent(event);
+      })
+      .catch(error => {
+        throw stackError("Failed to track click", error);
+      });
```

Comment:
This is an interesting case where if the user opted into no purposes or opted out of all purposes, then click tracking will fail and the error will go unhandled, since the click tracking isn't actuated by the customer calling a command. Without adding context to the error, the error description was pretty generic (like "The user opted into no purposes").

Reply:
I'm still thinking about how to best handle this scenario. Because the error doesn't bubble up through generic error massaging logic, the error says nothing about Alloy and the error can't be silenced by `errorsEnabled: false`. I've logged this here: https://jira.corp.adobe.com/browse/CORE-39185